### PR TITLE
Introduce an `interval` field to capture workflows

### PR DESCRIPTION
**Pull request by kiahna-tucker**
## Issues

The issues directly below are advanced by this PR:
https://github.com/estuary/ui/issues/1194

### 1194

The following features are included in this PR:

* Simplify the logic for the component that exposes the `interval` property in capture workflows.

### Manually tested

Approaches to testing are as follows:

* Validate that the capture interval section does not appear in materialization workflows.

* Validate that the capture interval section does not appear in capture workflows when the selected connector does not specify a default capture interval.

* Validate that the field is prefilled with the default interval in the capture create workflow when the selected connector specifies one.

* Validate that the field is prefilled with the default interval in the capture edit workflow when the live specification does not specify an interval but a default capture interval exists.

* Validate that the field is prefilled with the interval specified in the live specification in the capture edit workflow when it exists.

* Validate that the field considers input in an **unsupported** interval format invalid.

### Automated tests

Test coverage updated for the following utils:

* `formatCaptureInterval`

#### Playwright tests ran locally

-   [ ] Admin
-   [ ] Captures
-   [ ] Collections
-   [ ] HomePage
-   [ ] Login
-   [ ] Materialization

## Screenshots

**Capture interval field | Value selected**

pr_screenshot-1358-capture_interval-standard-single_unit_interval

<br />
<br />

**Capture interval field | Tooltip**

pr_screenshot-1319-capture_interval-standard-tooltip

<br />
<br />

**Capture interval field | Predefined interval options**

pr_screenshot-1358-capture_interval-standard-menu-options

<br />
<br />

**Capture interval field | Empty**

pr_screenshot-1358-capture_interval-standard-empty

<br />
<br />

**Capture interval field-level error**

pr_screenshot-1358-capture_interval-standard-error-input-compound

### CODE REVIEW

File: src/components/capture/Interval/index.tsx [hunk 1]
```diff
@@ -108,162 +78,69 @@ function CaptureInterval({ readOnly }: CaptureIntervalProps) {
...
+                    <TextField
+                        {...params}
+                        InputProps={{
+                            ...InputProps,
+                            sx: { borderRadius: 3 },
+                        }}
+                        error={errorsExist}
+                        label={label}
+                        size="small"
+                        variant="outlined"
```
Comment (travjenkins) [inline src/components/capture/Interval/index.tsx:118]: This should probably match the other inputs and just use standard

File: src/components/capture/Interval/index.tsx [hunk 2]
```diff
@@ -108,162 +78,69 @@ function CaptureInterval({ readOnly }: CaptureIntervalProps) {
...
+            />
+
+            <FormHelperText style={{ marginLeft: 0 }}>
+                {intl.formatMessage(
+                    { id: 'captureInterval.input.description' },
+                    {
+                        value: Duration.fromObject(defaultInterval).toHuman({
+                            listStyle: 'long',
+                            unitDisplay: 'short',
+                        }),
```
Comment (travjenkins) [inline src/components/capture/Interval/index.tsx:132]: Since `toHuman' does not filter out `0` let's go ahead and do it ourselves. It is really weird to see 0 hours, n minutes, 0 seconds

File: src/utils/time-utils.ts [hunk 3]
```diff
@@ -1,105 +1,26 @@
...
-        if (minutes > 0) {
-            segments.push(`${minutes}m`);
-        }
-
-        if (seconds > 0) {
-            segments.push(`${seconds}s`);
-        }
+        formattedInterval = Duration.fromObject(
+            parsePostgresInterval(interval)
+        ).toFormat("h'h' m'm' s's'");
```
Comment (travjenkins) [inline src/utils/time-utils.ts:21]: Weird - but can you store this in a string template. That way we can put whatever in the format pattern.

File: src/utils/__tests__/time-utils.test.ts
Comment (travjenkins) [inline src/utils/__tests__/time-utils.test.ts:1]: Add a few more edge cases - that way we have some documentation that calls out we allow really really really huge numbers in here.

File: src/components/capture/Interval/index.tsx [hunk 4]
```diff
@@ -57,31 +41,17 @@ function CaptureInterval({ readOnly }: CaptureIntervalProps) {
...
-            unit === 'i'
-                ? intervalErrorsExist
-                : intervalErrorsExist && !NUMERIC_RE.test(input)
-        );
-    }, [input, unit]);
+    const errorsExist = useMemo(
+        () =>
+            Boolean(
+                input && hasLength(input) && !CAPTURE_INTERVAL_RE.test(input)
+            ),
```
Comment (travjenkins) [inline src/components/capture/Interval/index.tsx:52]: We have basically the same test in 3 places - we should just make this a general function in the time utils.

File: src/components/capture/Interval/index.tsx (hunk 1 above)
Comment (kiahna-tucker) [reply src/components/capture/Interval/index.tsx:118]: I am not a fan of the look of the standard input here. Dave will likely express the sentiment that the field does not appear to be interactive or is easily overlooked, but I will update it since that is your preference.

File: src/utils/__tests__/time-utils.test.ts
Comment (kiahna-tucker) [reply src/utils/__tests__/time-utils.test.ts:1]: The testing library does restrict the size of the numbers that can be tested here, but I can extend the tests to sit just before the limitation. I can also note this in a comment.

File: src/components/capture/Interval/index.tsx (hunk 4 above)
Comment (kiahna-tucker) [reply src/components/capture/Interval/index.tsx:52]: Sweeping through the code to see where this condition appears three times. As of right now, I do not see where this is occurring.

File: src/stores/Binding/Store.ts [hunk 5]
```diff
@@ -1024,8 +1021,18 @@ const getInitialState = (
...
+                    const intervalObject =
                         Duration.fromISOTime(defaultInterval).toObject();
+
+                    Object.entries(intervalObject).forEach(
+                        ([timeUnit, unitValue]) => {
+                            if (unitValue === 0) {
+                                intervalObject[timeUnit] = undefined;
+                            }
+                        }
+                    );
```
Comment (travjenkins) [inline src/stores/Binding/Store.ts:1033]: These should be filtered out in the format function and not the store. I know this means we'll run the filter more but it isn't that huge of a piece of logic.

Comment (kiahna-tucker) [reply src/stores/Binding/Store.ts:1033]: The format function does not alter the default interval, but I can look into rearranging things to make that work.

Comment (travjenkins) [reply src/stores/Binding/Store.ts:1033]: Just make it a boolean option so we can control it just for the default output.

The goal is to have this logic where it should be eventually when luxon does add this feature : https://github.com/moment/luxon/issues/1134

Comment (travjenkins) [reply src/stores/Binding/Store.ts]: We could also just make a new function just for the defaulted option... since it is a special case and could be easier to understand than an extra boolean.

**Review by travjenkins — APPROVED** [review]
lgtm - didn't manually test the last round of changes
